pkg/motor/connect.go

```diff
@@ -0,0 +1,206 @@
+package motor
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/libp2p/go-libp2p-core/network"
+	"github.com/libp2p/go-libp2p-core/peer"
+	"github.com/libp2p/go-libp2p/p2p/protocol/circuitv2/proto"
+	"github.com/libp2p/go-msgio"
+	ma "github.com/multiformats/go-multiaddr"
+	"github.com/pkg/errors"
+	"github.com/sonr-io/sonr/pkg/config"
+	"github.com/sonr-io/sonr/pkg/host"
+	dp "github.com/sonr-io/sonr/pkg/motor/x/discover"
+	tp "github.com/sonr-io/sonr/pkg/motor/x/transmit"
+	ct "github.com/sonr-io/sonr/third_party/types/common"
+	mt "github.com/sonr-io/sonr/third_party/types/motor/api/v1"
+	v1 "github.com/sonr-io/sonr/third_party/types/motor/api/v1/service/v1"
+)
+
+func (mtr *motorNodeImpl) Connect(request mt.ConnectRequest) (*mt.ConnectResponse, error) {
+	if mtr.SonrHost != nil {
```

Comment:
would there be a case when we would want to reconnect to a host after already establishing one?

Reply:
Good point. I'm adding a param to `ConnectRequest` which will allow you to skip this check explicitly.